### ISSUE TITLE
Display of PHP function params doesn't reset an untyped param

### ISSUE THREAD

**Issue by lkppo** [NaturalDocs/NaturalDocs#50]
Hello,

I do some test of ND for a PHP project.

I wrote a simple method description like this:

```
  /*
       Method: testConnection
     
       Test connection to a MySQL/MariaDB server.
     
       Parameters:
       
          $dbhost - String: Hostname/IP of the server and it's port (optional, eg. hostname:3306)
          $dbname - String: Name of the database
          $dbuser - String: Login
          $dbpwd - String: Password
          $dbpers - Bool: Open a persistent connection if TRUE
       
       Returns:
       
          Array of one entry where the key is a boolean and the value an error message.
    */
    function testConnection(string $dbhost, $dbname, $dbuser, bool $dbpwd, $dbpers)
    {
        $dbhost = $dbpers === TRUE ? 'p:' . $dbhost : $dbhost;
        $h = mysqli_connect($dbhost, $dbuser, $dbpwd, $dbname);
        if (mysqli_connect_errno() == 0) {
            mysqli_close($h);
            return [TRUE => ""];
        } else {
            return [FALSE => mysqli_connect_error()];
        }
    }
```

ND seems to annotate each param with the type I give it first and keep going with it when I don't define one until it meet a new type definition. For a language like PHP this is not good because the type of a parameter is optional.

Am I misusing ND, do I have to mention all types or none, or is this an error?

I also don't understand why it displays $ after the type in red.

![testConnection](https://user-images.githubusercontent.com/10128284/94500779-73251e00-0200-11eb-90b8-eabf22ab5590.png)

-- 
Stéphane Aulery

**Comment by NaturalDocs** [comment]
This is an issue with Natural Docs' generalized prototype parsing not working well with PHP's conventions.  I added some special handling for PHP which will be included in the next release, which will probably be put out this week.

**Comment by lkppo** [comment]
Good news, thank you.

**Comment by NaturalDocs** [comment]
The fix for this is included in [Natural Docs 2.1 Beta 1](https://www.naturaldocs.org/download/#Development).

**Comment by lkppo** [comment]
Le 06/10/2020 19:41, Greg Valure a écrit :

> The fix for this is included in Natural Docs 2.1 Beta 1 [1].

I travel this week. I will test that as soon as possible. Thank you.

-- 
Stéphane Aulery 

Links:
------
[1] https://www.naturaldocs.org/download/#Development
